app-gateway.tf

```diff
@@ -1,6 +1,21 @@
+locals {
+  prod_vault      = "${(var.env == "prod") ? "infra-vault-prod" : ""}"
+  nonprod_vault   = "${(var.env == "demo" || var.env == "aat") ? "infra-vault-nonprod" : ""}"
+  sandbox_vault   = "${(var.env == "sandbox" || var.env == "saat" || var.env == "sprod") ? "infra-vault-sandbox" : ""}"
+
+  vaultNameIfSubscriptionPresent = "infra-vault-${var.subscription}"
+
+  vaultName = "${var.subscription != "" ? local.vaultNameIfSubscriptionPresent : format("%s%s%s%s", local.prod_vault, local.nonprod_vault, local.sandbox_vault)}"
```

Comment:
you don't need this the subscription will always be present for you

Reply:
done thanks